### ISSUE TITLE
[Question] How do I persist user state (e.g. refreshing page) without service worker?

### ISSUE THREAD

**Issue by LuchoTurtle** [AxaGuilDEv/react-oidc#730]
### Issue and Steps to Reproduce

Before anything, huge thanks for the library, it is amazing and you should all be proud of it and how well you are maintaining it!

I have a question regarding persisting user state (i.e. user is still logged in) after refreshing a page. I've used your [demo](https://black-rock-0dc6b0d03.1.azurestaticapps.net/profile) and changed it locally to test. I only commented the configuration below:

```
    client_id: 'interactive.public.short', //  interactive.public.short
    redirect_uri: window.location.origin+'/authentication/callback', // http://localhost:4200/authentication/callback
    scope: 'openid profile email api offline_access',
    authority: 'https://demo.identityserver.io',
    refresh_time_before_tokens_expiration_in_second: 70,
    //service_worker_relative_url:'/OidcServiceWorker.js',                  // commented here
    //service_worker_only: false,                                           // commented here
```

With this, the demo should use the memory storage, which in fact works (the access token is not hidden by the service worker). 

![image](https://user-images.githubusercontent.com/17494745/158605713-27da625b-8449-495c-987d-e2f0547f5fa3.png)

However, when I refresh the page, the user is no longer authenticated 🤔 

![image](https://user-images.githubusercontent.com/17494745/158605992-73aba9bf-446f-4ce0-a010-738ca3b9de5f.png)

Keep in mind, if I had used the service worker, **it would have worked and persisted the user state between refreshes**. However, when I choose not to, it does not.

I was under the impression the access and ID tokens were saved in local storage, that way this doesn't happen. Am I missing something? How can I achieve normal behaviour (refreshing and user still be logged in) without using a service worker?

Thank you!

### Versions

Latest version

#### Expected
Expected user to not be asked to login again after refreshing page.

#### Actual
User loses "logged in" status and has to login again.

### Additional Details
None

**Comment by guillaume-chervet** [comment]
hi @LuchoTurtle , thank you for your issue.
We are using Session Storage that must be keep until your close your tabulation.

if you it F5 the data should be still here.
if you it ctrl+F5, it erase the data, it is a normal behavior for developers.

if you do not have this behavior, what is your browser?

**Comment by guillaume-chervet** [comment]
Service Worker mode is really more secure than the other mode.

**Comment by LuchoTurtle** [comment]
> Service Worker mode is really more secure than the other mode.

I fully agree! I'm inclusively using it right now 🙂! However, there was a situation that I could not and I tried to see if it worked without service worker.



> hi @LuchoTurtle , thank you for your issue. We are using Session Storage that must be keep until your close your tabulation.
> 
> if you it F5 the data should be still here. if you it ctrl+F5, it erase the data, it is a normal behavior for developers.
> 
> if you do not have this behavior, what is your browser?

If I comment those lines (not use the service worker), after logging in, if I F5, it shows me as unauthenticated and I have to login again (even by hard refreshing with CTRL + F5). I was using Chrome and I did this experiment with your demo 🙂.

I'm just wondering if this is expected. It works wonderfully with service worker though! If it is not, I'd be happy to try and make a PR for a feature like this for those who need it, I find your work amazing 🙏!
